sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -30,21 +31,34 @@ public abstract class TestEnvironment
 
         private readonly Dictionary<string, string> _environmentFile = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
-        protected TestEnvironment(string serviceName)
+        protected TestEnvironment()
         {
-            _prefix = serviceName.ToUpperInvariant() + "_";
             if (RepositoryRoot == null)
             {
                 throw new InvalidOperationException("Unexpected error, repository root not found");
             }
 
-            var sdkDirectory = Path.Combine(RepositoryRoot, "sdk", serviceName);
+            var testProject = GetType().Assembly.GetCustomAttributes<AssemblyMetadataAttribute>().Single(a => a.Key == "SourcePath").Value;
```

Comment:
Consider factoring this out into a shared method as it is also used in RecordedTestBase.